src/routes-data/sidebar-routes.jsx

```diff
@@ -0,0 +1,222 @@
+                // set the dropdown useState active.
+                setIsActive(!isActive);
+            }
+        }
+    }, []);
+    const handleToggleDropdown = () => {
+        setIsActive(!isActive);
+    };
+    return (
+        <div className={styles.dropdown}>
+            <div className={`${styles.dropdownBtn} ${isActive ? styles.boldText : ''}`} onClick={handleToggleDropdown}>
+                Implement Now
+                <span className={`${styles.arrow} ${isActive ? styles.down : ''}`} />
+            </div>
+            {isActive && (
+                <div className={styles.dropdownList}>
+                    {implement_now.map(items => (
+                        <Link
+                            key={items.path}
+                            to={'getting-started/implement-now/' + items.path}
+                            className={`${styles.dropdownContent} ${
+                                items.path === split_current_location[split_current_location.length - 1]
+                                    ? styles.selected
+                                    : ''
+                            }`}
+                        >
+                            {items.label}
+                        </Link>
+                    ))}
+                </div>
+            )}
+        </div>
+    );
+};
+
+function LinkComponent({ route, path }) {
+    const { pathname } = useLocation();
+    return (
+        <div key={route.label}>
+            {route.children ? (
+                <div className={styles.menuHeader}>{route.label}</div>
+            ) : (
+                <Link
+                    to={path}
+                    className={`${styles.menuItem} ${path === pathname.replace('/docs/', '') ? styles.selected : ''}`}
+                >
+                    {route.label}
+                </Link>
+            )}
+            {route.children &&
+                route.children.map(child => {
+                    // If there are children, recursively go over the nested children
+                    // till there are none anymore.
+                    return !child.is_collapsible ? (
+                        <React.Fragment>
+                            <LinkComponent route={child} path={`${path}/${child.path}`} />
+                        </React.Fragment>
+                    ) : (
+                        <React.Fragment>
+                            <ImplementDropdown route={child} />
+                        </React.Fragment>
+                    );
+                })}
+        </div>
+    );
+}
+
+export const SidebarMenuItems = ({ routes }) => {
+    const routes_array = Object.values(routes);
+    const map_links = Object.entries(routes_array).map(items => {
+        const route = items[1];
+        return (
+            <React.Fragment>
+                <LinkComponent route={route} path={route.path} key={route.path} />
+            </React.Fragment>
+        );
+    });
+    return map_links;
```

Comment:
h: this is the main sidebar component or a way to generate sidebar
a: is this a tree?
h: yes, gets routes tree
a: but <LinkComponent /> is then a component

Reply:
here I have the most concerns, because when vi read it then vi see that it is at the bottom of the PR - not on top of priority for the reader of the PR - only when reader is most tired you see the most important part
let's pls add refactor updates to the component as seperate PRs so we have some logic in small PR seperate from business data such as exact hierarchy of routes of the API app